vector/src/main/java/im/vector/app/core/utils/SystemSettingsProvider.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.utils
+
+import android.content.Context
+import android.provider.Settings
+import javax.inject.Inject
+
+/**
+ * A helper to get system settings.
```

Comment:
> does the consumer of getSystemFontScale benefit from knowing it's a System setting

Yes, the main purpose of this class is to provide system settings to be compared with app settings or used instead of it. So there is a clear and significant difference between system setting and app setting for consumer of this class

Reply:
> this might be more flexible if it was coupled to the font domain rather than SystemSettings, eg FontSettingsProvider

yes, it would be more flexible, but that's also a level of abstraction which we don't need and most likely we won't need in a future. There are no other system settings we provide for now and adding another class for font domain will just increase number of classes involved in resolving this setting, without giving us any real benefit from it's flexibility